src/lib.rs

```diff
@@ -82,15 +82,15 @@ pub fn scan<BdS: AsRef<Path>, BdI: IntoIterator<Item = BdS>, Sp: AsRef<Path>, As
         };
         for entry in dir_iter.flatten() {
             let fpath = entry.path();
-            let fname = match entry.file_name().into_string() {
-                Ok(n) => n,
-                _ => continue,
-            };
+            let fname = entry.file_name();
 
             // If hidden files not allowed, ignore dotfiles.
-            if ignore_dotfiles && fname.starts_with('.') {
+            // Rust RFC 900 &c.: there's no way to check if a Path/OsStr starts with a prefix;
```

Comment:
We can on Unix at least just compare the first bytes using https://doc.rust-lang.org/std/os/unix/ffi/trait.OsStrExt.html#tymethod.as_bytes, i.e. under some `#cfg[(unix)]` or so.  I'd honestly be a bit surprised if we ever had users on non-Unix platforms (but, I wouldn't say we should gratuitously not compile there - at least with Rust it's ergonomic to do conditional compilation and most high level abstractions peel away at compile time too).

Reply:
I considered this, but nothing else in the crate uses a system-specific API, so I went with this which decays graciously I think.